packages/contracts-bedrock/test/universal/BenchmarkTest.t.sol

```diff
@@ -30,23 +30,51 @@ contract SetPrevBaseFee_Test is CommonTest {
 
 contract GasBenchMark_L1CrossDomainMessenger is CommonTest {
     function test_sendMessage_benchmark_0() external {
+        // Skip if the test is running in coverage.
+        if (vm.isContext(VmSafe.ForgeContext.Coverage)) {
+            vm.skip(true);
+        }
+
+        // Setup
         vm.pauseGasMetering();
+        uint256 startingGas = gasleft();
         setPrevBaseFee(vm, address(optimismPortal2), 1 gwei);
         // The amount of data typically sent during a bridge deposit.
         bytes memory data =
             hex"ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff";
         vm.resumeGasMetering();
+
+        // Test
         l1CrossDomainMessenger.sendMessage(bob, data, uint32(100));
+
+        // Assert
+        vm.pauseGasMetering();
+        assert(startingGas - gasleft() < 400_000);
+        vm.resumeGasMetering();
     }
 
     function test_sendMessage_benchmark_1() external {
+        // Skip if the test is running in coverage.
+        if (vm.isContext(VmSafe.ForgeContext.Coverage)) {
+            vm.skip(true);
+        }
+
+        // Setup
         vm.pauseGasMetering();
+        uint256 startingGas = gasleft();
         setPrevBaseFee(vm, address(optimismPortal2), 10 gwei);
         // The amount of data typically sent during a bridge deposit.
         bytes memory data =
             hex"ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff";
         vm.resumeGasMetering();
+
+        // Test
         l1CrossDomainMessenger.sendMessage(bob, data, uint32(100));
+
+        // Assert
+        vm.pauseGasMetering();
+        assert(startingGas - gasleft() < 3_000_000);
```

Comment:
Same question about this 3M value. Can we pull these into constants with comment justifying their value?
```suggestion
        assertLt(startingGas - gasleft(), 3_000_000);
```

Reply:
ok I see there are a lot of different values, instead of constants in the contract let's just add a comment in each test explaining the value/requirement

Reply:
Spoke with @smartcontracts, these are arbitrary based on the previous snapshot values + some margin. We should have a separate follow up task to work with the protocol team and define proper gas-sensitive tests, including OPCM deploy/upgrade costs